pinot-core/src/main/java/com/linkedin/pinot/core/segment/creator/impl/SegmentIndexCreationDriverImpl.java

```diff
@@ -229,9 +229,9 @@ private void serializeTree(StarTreeBuilder starTreeBuilder) throws Exception {
     Map<String, HashBiMap<Object, Integer>> dictionaryMap = starTreeBuilder.getDictionaryMap();
     StarTree tree = starTreeBuilder.getTree();
     HashBiMap<String, Integer> dimensionNameToIndexMap = starTreeBuilder.getDimensionNameToIndexMap();
-    StarTreeIndexNode node = tree.getRoot();
+    StarTreeIndexNode node = (StarTreeIndexNode) tree.getRoot();
```

Comment:
In the first implementation, SegmentIndexCreationDriverImpl will always create the existing implementation, and then  we create the version from it. Will re-visit this in the next round of changes.